Sources/Topsort-Analytics/AuctionManager.swift

```diff
@@ -0,0 +1,61 @@
+import Foundation
+
+private let AUCTIONS_TOPSORT_URL = URL(string: "https://api.topsort.com/v2/auctions")!
+private let MAX_AUCTIONS = 5
+
+class AuctionManager {
+    public static let shared = AuctionManager()
+    private init() {
+        self.client = HTTPClient(apiKey: nil)
+    }
+    private var url: URL = AUCTIONS_TOPSORT_URL
+    private var client: HTTPClient
+
+    public func configure(apiKey: String, url: String?) {
+        self.client.apiKey = apiKey
+        if let url = url {
+            guard let url = URL(string: "\(url)/auctions") else {
+                fatalError("Invalid URL")
+            }
+            self.url = url
+        }
+    }
+
+    public func executeAuctions(auctions: [Auction]) async -> AuctionResponse? {
+        if auctions.count > MAX_AUCTIONS {
+            print("Auctions count exceeds the limit of \(MAX_AUCTIONS)")
+            return nil
```

Comment:
I don't really like the design of printing the error and then returning nil, but that comes from the previous manager, so we can keep it with the promise of changing it later.

Reply:
agree, I could make a refactor later